File: Resueltos_por_competidor/Juan_Martinez/Game23.cpp
/*
 * Autor: Juan Martinez
 * Problema: Game 23 (1141A)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/1141/A
 * Difficulty: 1000
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int ll;

bool b = false;
ll res;
void calcular(ll n, ll m, ll c) {
    if(n >= m) {
        if(n == m) b = true, res = c;
    } else {
        calcular(n*3, m, c+1);
        calcular(n*2, m, c+1);
    }
}
int main() {
    ll n, m;
    cin>>n>>m;
    calcular(n, m, 0);
    if(!b)cout<<-1<<endl;
    else cout<<res<<endl;
}

File: Resueltos_por_competidor/Martin_Peña/Game23.java
/*
* Autor: pgms1103-afk
* Problema: A. Game 23
* Juez Online: Codeforces
* Veredicto: Accepted
* URL: https://codeforces.com/problemset/problem/1141/A	
*/
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Game23 {

	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

	public static void main(String[] args) throws IOException {

		String datos[] = br.readLine().split(" ");
		long numeroPrincipio = Integer.parseInt(datos[0]);
		long numeroFinal = Integer.parseInt(datos[1]);

		if(numeroFinal % numeroPrincipio != 0) {
			System.out.println(-1);
			return;
		}
		long resultado = numeroFinal/numeroPrincipio;
		//System.out.println(resultado);
		long tres = 0;
		long dos = 0;
		
		while(resultado % 3 == 0) {
			tres++;
			resultado /= 3;
			//System.out.println(resultado);
		}
		while(resultado % 2 == 0) {
			dos++;
			resultado /= 2;
			//System.out.println(resultado);
		}
		
		if(resultado == 1) {
			System.out.println(dos+tres);
		}else {
			System.out.println(-1);
		}
	}

}
